src/summarycode/file_cat.py

```diff
@@ -0,0 +1,1265 @@
+        """
+        Return True if this Categorizer applies to this resource or False or None
+        otherwise.
+        """
+        raise NotImplementedError
+
+
+class ArchiveAndroid(Categorizer):
+    order = 10
+    analysis_priority = "1"
+    file_category = "archive"
+    file_subcategory = "Android"
+    rule_applied = "ArchiveAndroid"
+
+    @classmethod
+    def categorize(cls, resource):
+        return extension_in(resource.extension, [".apk", ".aar"])
+
+
+class ArchiveDebian(Categorizer):
+    order = 10
+    analysis_priority = "1"
+    file_category = "archive"
+    file_subcategory = "debian"
+    category_notes = "special type of archive"
+    rule_applied = "ArchiveDebian"
+
+    @classmethod
+    def categorize(cls, resource):
+        return extension_in(resource.extension, [".deb"])
+
+
+class ArchiveGeneral(Categorizer):
+    order = 10
+    analysis_priority = "1"
+    file_category = "archive"
+    file_subcategory = "general"
+    rule_applied = "ArchiveGeneral"
+
+    @classmethod
+    def categorize(cls, resource):
+        return extension_in(
+            resource.extension,
+            [
+                ".7zip",
+                ".bz",
+                ".bz2",
+                ".bzip",
+                ".gz",
+                ".gzi",
+                ".tar",
+                ".tgz",
+                ".xz",
+                ".zip",
+            ],
+        )
+
+
+class ArchiveIos(Categorizer):
+    order = 10
+    analysis_priority = "1"
+    file_category = "archive"
+    file_subcategory = "iOS"
+    rule_applied = "ArchiveIos"
+
+    @classmethod
+    def categorize(cls, resource):
+        return extension_in(resource.extension, [".ipa"])
+
+
+class ArchiveRpm(Categorizer):
+    order = 10
+    analysis_priority = "1"
+    file_category = "archive"
+    file_subcategory = "rpm"
+    category_notes = "special type of archive"
+    rule_applied = "ArchiveRpm"
+
+    @classmethod
+    def categorize(cls, resource):
+        return extension_in(resource.extension, [".rpm"])
+
+
+class BinaryAr(Categorizer):
+    order = 10
+    analysis_priority = "1"
+    file_category = "binary"
+    file_subcategory = "ar"
+    rule_applied = "BinaryAr"
+
+    @classmethod
+    def categorize(cls, resource):
+        return resource.mime_type in ["application/x-archive"]
+
+
+class BinaryElfExec(Categorizer):
+    order = 10
+    analysis_priority = "1"
+    file_category = "binary"
+    file_subcategory = "elf-exec"
+    rule_applied = "BinaryElfExec"
+
+    @classmethod
+    def categorize(cls, resource):
+        return resource.mime_type in ["application/x-executable"]
+
+
+class BinaryElfKo(Categorizer):
+    order = 10
+    analysis_priority = "1"
+    file_category = "binary"
+    file_subcategory = "elf-ko"
+    rule_applied = "BinaryElfKo"
+
+    @classmethod
+    def categorize(cls, resource):
+        return resource.mime_type in ["application/x-object"] and extension_startswith(
```

Comment:
Why startswith here? Do we have examples of LKMs that do not have a full .ko extension?